### ISSUE TITLE
operator IstioControlPlane k8s:affinity:  mapped to wrong level

### ISSUE THREAD

**Issue by kramvan1** [istio/istio#18130]
**Bug description**

Example from Doc: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#an-example-of-a-pod-that-uses-pod-affinity

The mapping https://github.com/istio/operator/blob/master/data/translateConfig/translateConfig-1.4.yaml#L20-L22  appears to be wrong.

fails for the follow components:
- configManagement
  - galley
- gateways
  - ingress
  - egress
- security
  - citadel
- telemetry
  - telemetry
- trafficManagement
  - pilot

**Expected behavior**

The `k8s:Affinity:`  settings to be applied to the pod.

The mapping should be for the Deployments/Pods to get the correct Affinity settings.
```
kubernetesMapping:
  "{{.FeatureName}}.Components.{{.ComponentName}}.K8S.Affinity":
    outPath: "[{{.ResourceType}}:{{.ResourceName}}].spec.template.spec.affinity"
```

**Steps to reproduce the bug**

CR:
```
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
metadata:
  name: managed-istiocontrolplane
  namespace: istio-operator
spec:
  configManagement:
    components:
      galley:
        enabled: true
        k8s:
          affinity:
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
              - podAffinityTerm:
                  labelSelector:
                    matchExpressions:
                    - key: app
                      operator: In
                      values:
                      - galley
                  topologyKey: kubernetes.io/hostname
```

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

latest operator master.

**How was Istio installed?**

operator controller

**Environment where bug was observed (cloud vendor, OS, etc)**

ibm public cloud

**Comment by linsun** [comment]
@kramvan1 thanks for reporting this, will check

**Comment by linsun** [comment]
also fixed in 1.4 as well https://github.com/istio/operator/pull/450
